### PR TITLE
feat(sdk): support saml connection

### PR DESCRIPTION
**Pull request by jackworkgit**

### CODE REVIEW

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java [hunk 1]
```diff
@@ -79,6 +80,17 @@ public Mono<HttpResponse<OrganizationConnection>> createConnection(
       @PathVariable String orgId,
       @RequestBody CreateConnectionDto request,
       JwtAuthenticationToken authenticationToken) {
+    if (request.getStrategy().equals(ConnectionStrategryEnum.SAML)) {
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java:83]: move if ... else .. to OrganizationService.createConnection(...)

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java [hunk 2]
```diff
@@ -143,4 +155,19 @@ public Mono<HttpResponse<Paging<Role>>> listRoles(
...
+
+  @Operation(summary = "Update a connection")
+  @PatchMapping("/{orgId}/connections")
+  public Mono<HttpResponse<OrganizationConnection>> updateConnection(
+      @PathVariable String orgId,
+      @RequestBody UpdateConnectionDto updateConnectionDto,
+      JwtAuthenticationToken authenticationToken) {
+    return Mono.just(
+        HttpResponse.ok(
+            service.updateSAML(
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java:167]: The update code looks only support SAML, but creationg support OIDC , DB and SAML

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/CreateConnectionDto.java [hunk 3]
```diff
@@ -1,35 +1,13 @@
...
-      Map<String, Object> map = new HashMap<>();
-      map.put("client_id", getClientId());
-      map.put("discovery_url", getDiscoveryUrl());
-      map.put("scope", getScope());
-      map.put("type", "front_channel");
-      return map;
-    }
-  }
+  private OidcConnection odic;
+  private SamlConnection samlConnection;
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/CreateConnectionDto.java:12]: rename `samlConnection` to `saml`

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/SamlConnection.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.consoleconnect.vortex.iam.dto;
+
+import java.util.Map;
+import lombok.Data;
+
+@Data
+public class SamlConnection {
+  private ConnectionOptions options;
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/SamlConnection.java:8]: Looks like you can move all properties from COnnectionOptions to SamlConnection class, and delete ConnectionOptions

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+@Data
+public class UpdateConnectionDto {
+
+  @NotBlank private String id;
+
+  private ConnectionStrategryEnum strategy = ConnectionStrategryEnum.SAML;
+
+  private OidcConnection odic;
+
+  private SamlConnection samlConnection;
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/dto/UpdateConnectionDto.java:16]: rename `samlConnection` to `saml`

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/enums/ConnectionStrategryEnum.java [hunk 6]
```diff
@@ -2,8 +2,9 @@
 
 public enum ConnectionStrategryEnum {
   OIDC("oidc"),
-  SAML("saml"),
-  OAUTH2("oauth2");
+  SAML("samlp"),
+  OAUTH2("oauth2"),
+  DB("auth0");
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/enums/ConnectionStrategryEnum.java:7]: maybe rename `DB` to `AUTH0` ?  a bit confuse mapping DB to auth0.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/model/AttributeProperty.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package com.consoleconnect.vortex.iam.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+@Data
+public class AttributeProperty {
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/model/AttributeProperty.java:7]: What's the purpose of this class ? from the name , not sure what's the purpose of this class.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/ConnectionService.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class ConnectionService {
+
+  @Async
+  public void cleanConnectionAndMembers(
+      ManagementAPI managementAPI,
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/ConnectionService.java:19]: looks like you want to delete the following data from this org:
1. all members
2. all pending invitations
3. all  connections

Pls check if the function only accept the parameter `orgId` should be ok.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 9]
```diff
@@ -83,8 +95,10 @@ public Organization update(String orgId, UpdateOrganizationDto request, String c
       throw VortexException.badRequest("Payload cannot be empty.");
     }
     try {
-      Organization organization = findOne(orgId);
+      OrganizationsEntity organizationsEntity = this.auth0Client.getMgmtClient().organizations();
+      findOrganization(orgId, organizationsEntity);
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:99]: Pls check why the return of  `findOrganization` won't be assigned to only varirable

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 10]
```diff
@@ -98,11 +112,29 @@ public Organization update(String orgId, UpdateOrganizationDto request, String c
...
+        metadata.put(META_STATUS, request.getStatus());
       }
-      OrganizationsEntity organizationsEntity = this.auth0Client.getMgmtClient().organizations();
+
       Request<Organization> organizationRequest = organizationsEntity.update(orgId, organization);
-      return organizationRequest.execute().getBody();
+      Organization response = organizationRequest.execute().getBody();
+
+      //
+      if (organization.getMetadata().get(META_STATUS).equals(OrgStatusEnum.ACTIVE.name())
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:122]: looks like you missed the logic:
metadata.status == INACTIVE && request.status == ACTIVE

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 11]
```diff
@@ -331,25 +371,173 @@ public OrganizationConnection createConnection(
...
+      return bindOrganizationConnection(
+          orgId, createdConnection, organizationsEntity, LoginTypeEnum.SSO);
+
+    } catch (Auth0Exception e) {
+      log.error("create connections.error", e);
+      throw VortexException.internalError("Failed to create connections of organization: " + orgId);
+    }
+  }
+
+  public OrganizationConnection dbConnection(String orgId, String requestedBy) {
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:383]: rename to `createDBConnection`

**Review by DaveXiong — CHANGES_REQUESTED** [review]
comments added

**Comment by DaveXiong** [comment]
And pls fix the issues reported by SonarCloud

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/enums/ConnectionStrategryEnum.java (hunk 6 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/enums/ConnectionStrategryEnum.java:7]: OK

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java (hunk 1 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java:83]: ok

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/model/AttributeProperty.java (hunk 7 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/model/AttributeProperty.java:7]: To wrap Auth0 attributes. Or just use Map<String, Object>, there are a few fields.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/ConnectionService.java (hunk 8 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/ConnectionService.java:19]: Only one `orgId` may have issues. Old connections should be passed in this method.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 10 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:122]: The code is formatted to a new line. My logic is above.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 9 above)
Comment (ahasgx) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:99]: 1. Check organization exists before updating an organization，
2. updating an organization cannot use the variable organization  which queries from auth0, so  not assigned to variable

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java (hunk 10 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:122]: Here, the connection needs to be deleted to change the state from ACTIVE to INACTIVE.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java (hunk 2 above)
Comment (jackworkgit) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/controller/MgmtOrganizationController.java:167]: Now I think we can only oidc/saml updating. DB is created with no more parameters.

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 12]
```diff
@@ -306,8 +325,15 @@ public Paging<OrganizationConnection> listConnections(String orgId, int page, in
   public OrganizationConnection createConnection(
       String orgId, CreateConnectionDto request, String requestedBy) {
     log.info("creating connection:orgId:{}, {},requestedBy:{}", orgId, request, requestedBy);
-    try {
+    if (request.getStrategy().equals(ConnectionStrategryEnum.SAML)) {
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:328]: better to add more logic there:

```
if (request.getStrategy() == ConnectionStrategryEnum.SAML) {
      return createSAMLConnection(orgId, request.getSaml());
    } else if (request.getStrategy() == ConnectionStrategryEnum.OIDC) {
      return createOidcConnection(orgId, request);
    } else {
      throw VortexException.badRequest("Invalid connection strategy");
    }
```

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 13]
```diff
@@ -331,25 +357,166 @@ public OrganizationConnection createConnection(
...
       throw VortexException.internalError("Failed to create connections of organization: " + orgId);
     }
   }
+
+  private OrganizationConnection createSAMLConnection(String orgId, SamlConnection samlConnection) {
+    try {
+      ManagementAPI managementAPI = this.auth0Client.getMgmtClient();
+      Organization organization = managementAPI.organizations().get(orgId).execute().getBody();
+      Map<String, Object> metaData =
+          JsonToolkit.createObjectMapper().convertValue(samlConnection.getOptions(), Map.class);
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java]: Pls check the following code inspection description:
```
Unchecked assignment: 'java. util. Map' to 'java. util. Map<java. lang. String,java. lang. Object>' 
```

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 14]
```diff
@@ -331,25 +357,166 @@ public OrganizationConnection createConnection(
...
+    return updateSAMLConnection(orgId, request.getId(), request.getSaml());
+  }
+
+  private OrganizationConnection updateSAMLConnection(
+      String orgId, String connectionId, SamlConnection samlConnection) {
+    try {
+      ManagementAPI managementAPI = this.auth0Client.getMgmtClient();
+      Organization organization = managementAPI.organizations().get(orgId).execute().getBody();
+      Map<String, Object> metaData =
+          JsonToolkit.createObjectMapper().convertValue(samlConnection.getOptions(), Map.class);
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java]: Pls check the following code inspection description:

Unchecked assignment: 'java. util. Map' to 'java. util. Map<java. lang. String,java. lang. Object>'

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/ConnectionService.java (hunk 8 above)
Comment (DaveXiong) [reply vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/ConnectionService.java:19]: the above business logic will delete all members and invitations inside the organization, but only delete the specific connection, Does it make sense?  Looks like you would like to delete all data inside the organization ?

File: vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java [hunk 15]
```diff
@@ -331,25 +359,165 @@ public OrganizationConnection createConnection(
...
+    }
+  }
+
+  public OrganizationConnection updateConnection(
+      String orgId, UpdateConnectionDto request, String requestedBy) {
+    log.info("Updating connection:orgId:{}, {},requestedBy:{}", orgId, request, requestedBy);
+    if (request.getStrategy().equals(ConnectionStrategryEnum.OIDC)) {
+      return updateOidcConnection(orgId, request.getId(), request.getOdic());
+    }
+    return updateSAMLConnection(orgId, request.getId(), request.getSaml());
```
Comment (DaveXiong) [inline vortex-java-sdk/vortex-java-sdk-iam/src/main/java/com/consoleconnect/vortex/iam/service/OrganizationService.java:421]: Pls add the logic as following
```
if (request.getStrategy() == ConnectionStrategryEnum.SAML) {
      
    } else if (request.getStrategy() == ConnectionStrategryEnum.OIDC) {
      
    } else {
      throw VortexException.badRequest("Invalid connection strategy");
    }
```

**Review by DaveXiong — APPROVED** [review]
LGTM

**Pull request merged**